mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -258,24 +518,44 @@ void mav_vehicle::send_heartbeat()
 
     // Encode and send
     uint8_t mav_data_buffer[defaults::send_buffer_len];
-    mavlink_msg_heartbeat_encode(defaults::system_id, defaults::component_id,
+    mavlink_msg_heartbeat_encode(this->system_id, defaults::component_id,
                                  &mav_msg, &mav_heartbeat);
     int n = mavlink_msg_to_send_buffer(mav_data_buffer, &mav_msg);
 
     // Update timestamp
     cmd_custom_timestamps[cmd] = system_clock::now();
 
     if (send_data(mav_data_buffer, n) == -1) {
-        std::perror("[mav_vehicle] Error sending heartbeat");
+        print_verbose("Error sending heartbeat\n");
     }
 }
 
 void mav_vehicle::set_mode(mode m)
 {
+    set_mode(m, request_intervals_ms::set_mode);
+}
+
+void mav_vehicle::set_mode(mode m, int timeout)
+{
     using namespace std::chrono;
-    cmd_custom cmd = cmd_custom::SET_MODE;
+    cmd_custom cmd = cmd_custom::SET_MODE_GUIDED;
```

Comment:
We should not set any mode if set_mode() is called with an invalid mode. I think it's better to keep it as is: printing a warning message and not setting any mode.